ratelimit_test.go

```diff
@@ -7,10 +7,105 @@ import (
 
 	"go.uber.org/atomic"
 
-	"github.com/benbjohnson/clock"
 	"github.com/stretchr/testify/assert"
 )
 
+// testTime is a fake time used for testing.
+type testTime struct {
```

Comment:
If this is copied over from https://github.com/golang/time/commit/579cf78fd858857c0d766e0d63eb2b0ccf29f436, could we:
(1) put this is a separate file
(2) attribute the source clearly
(3) call out differences in the implementation? For example, I see that `advanceUnlocked` is different than the reference implementation, but I cannot easily tell why.

Reply:
done